### ISSUE TITLE
Duplicating Entry Doesn't Create a Unique UID

### ISSUE THREAD

**Issue by TonyDeStefano** [craftcms/cms#4653]
### Description

When you select an Entry and choose "Duplicate" from the action menu, it creates a new entry with the same UID.

### Steps to reproduce

1. Create an entry
2. Check the box next to the entry
3. Choose "Duplicate"

This query shows current duplicates in the database:

```sql
SELECT uid, count(*) AS `count` 
FROM elements 
GROUP BY uid 
HAVING `count` > 1
```

### Additional info

- Craft version: 3.1.34.2
- PHP version: 7.2.11

**Comment by TonyDeStefano** [comment]
"Save as New Entry" does the same thing. We end up with two elements that have the same UID.

**Comment by brandonkelly** [comment]
I had noticed this bug during 3.2 development, but thought it was limited to 3.2 alpha releases, so while 3.2 has a migration to fix duplicate UIDs, it only runs if you’re coming from 3.2.0-alpha.6 or later.

So I just added a new console command that you will be able to run once Craft 3.2.8 is released, which will run the same migration code.

```bash
./craft utils/fix-element-uids
```

To get the fix early, change your `craftcms/cms` requirement in composer.json to:

```json
"require": {
  "craftcms/cms": "dev-develop#0e0c89e4684e07d784a1e4d5bdcaa3395ee07a54 as 3.2.7",
  "...": "..."
}
```

Then run `composer update`.

**Comment by TonyDeStefano** [comment]
We are unable to upgrade to 3.2 right now. I submitted a PR for 3.1.

**Comment by brandonkelly** [comment]
We aren’t releasing additional 3.1 releases. You’ll just have to point your local install to your fork with the fix, until you are ready to update to 3.2.

**Comment by ryanpcmcquen** [comment]
@brandonkelly, this is a critical bug, hopefully you'll reconsider implementing some fix for this on `3.1`.

**Comment by brandonkelly** [comment]
Craft 3.1 isn’t an LTS release. As said you are welcome to use a fork with your own fix until you’re ready to update to 3.2.
